tests/integration/test_integration_TMP.py

```diff
@@ -0,0 +1,73 @@
+import io
+import os
+import pytest
+import time
+from main import generate_parser, get_gpu, run_extract_data, upload
+
+
+parser = generate_parser()
+test_folders = [entries for entries in os.listdir("tests/") if os.path.isdir(f"tests/{entries}")]
+for fold in set(test_folders):
+    if "baseboard.txt" not in os.listdir(f"tests/{fold}"):
+        test_folders.remove(fold)
+
+
+@pytest.fixture(params=test_folders)
+def args(request, monkeypatch):
+    args = parser.parse_args(['-f', f'tests/{request.param}'])
+    # the following function could prompt a menu requesting gpu location.
+    # Setting a value into stdin buffer just in case input is needed
+    cli_input = io.StringIO('b')
+    monkeypatch.setattr('sys.stdin', cli_input)
+    get_gpu(args)
+    return args
+
+
+def test_get_gpu(args):  # checking that gpu_location is correctly read
+    # checking mutual exclusion
+    assert any([args.cpu, args.gpu, args.motherboard]) \
+           and len([val for val in (args.cpu, args.gpu, args.motherboard) if val is True]) <= 1
```

Comment:
Or more synthetically:
```suggestion
    assert [args.cpu, args.gpu, args.motherboard].count(True) <= 1
```
But my question is: isn't this testing argparse, which is not really our job? Or am I missing something?

Reply:
Well, what I'm really testing it's the get_gpu function used in the fixture. That function read from gpu_location.txt and if there's no information, an input on gpu's location is requested. These operations are modifying the parameters parsed by argparse. So basically I'm assuring that get_gpu() doesn't mess up with parameters